src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -1009,6 +1009,31 @@ namespace AppInstaller::Repository
         return result;
     }
 
+    std::shared_ptr<ISource> CreateCompositeSource(
+        const std::shared_ptr<ISource>& installedSource,
+        const std::vector<std::shared_ptr<ISource>>& availableSources,
+        CompositeSearchBehavior searchBehavior)
+    {
+        if (availableSources.empty())
+        {
+            THROW_HR(E_UNEXPECTED);
+        }

```

Comment:
This seems like it should be fine to allow to happen, even if it is useless.  If it were to fail, it should be invalid arg.

Reply:
Allowed it.